apps/src/lib/util/speech.js

```diff
@@ -22,38 +20,153 @@ export function textToSpeech(
+  } else {
+    language = 'English';
+    gender = 'female';
+  }
+
+  const cachedBytes = Sounds.getSingleton().getTextBytes(language, text);
+
+  if (cachedBytes !== null && cachedBytes['hasProfanity']) {
+    outputWarning(
+      i18n.textToSpeechProfanity({
+        profanityCount: cachedBytes['profaneWords'].length,
+        profaneWords: cachedBytes['profaneWords'].join(', ')
+      })
+    );
+    return;
+  }
+  const asyncOutputWarning = getAsyncOutputWarning();
+  // There is no way to make ajax requests from html on the filesystem.  So
+  // the only way to play sounds is using HTML5. This scenario happens when
+  // students export their apps and run them offline. At this point, their
+  // uploaded sound files are exported as well, which means varnish is not
+  // an issue.
+  const forceHTML5 = window.location.protocol === 'file:';
+  if (cachedBytes !== null && cachedBytes[gender] !== undefined) {
+    if (cachedBytes[gender].length > 0) {
+      Sounds.getSingleton().addPromiseToSpeechQueue(
+        cachedBytes[gender].slice(0),
+        {
+          volume: 1.0,
+          loop: false,
+          forceHTML5: forceHTML5,
+          allowHTML5Mobile: true,
+          onEnded: () => {
+            Sounds.getSingleton().onSpeechFinished();
+          }
+        }
+      );
+    } else {
+      Sounds.getSingleton().addPromiseToSpeechQueue(
+        null,
+        {
+          volume: 1.0,
+          loop: false,
+          forceHTML5: forceHTML5,
+          allowHTML5Mobile: true,
+          onEnded: () => {
+            Sounds.getSingleton().onSpeechFinished();
+          }
+        },
+        {
+          language: language,
+          text: text,
+          gender: gender,
+          profanityFoundCallback: profaneWords => {
+            asyncOutputWarning(
+              i18n.textToSpeechProfanity({
+                profanityCount: profaneWords.length,
+                profaneWords: profaneWords.join(', ')
+              })
+            );
+          }
+        }
+      );
+    }
+  } else {
+    let ssml = `<speak version="1.0" xmlns="https://www.w3.org/2001/10/synthesis" xml:lang="en-US"><voice name="${voice}">${text}</voice></speak>`;
+    const url = `https://${region}.tts.speech.microsoft.com/cognitiveservices/v1`;
+
+    var request = new XMLHttpRequest();
+    const resultPromise = new Promise(async function(resolve, reject) {
+      let profaneWords = await findProfanity(text, languageCode);
+      if (profaneWords && profaneWords.length > 0) {
+        if (!Array.isArray(profaneWords)) {
+          profaneWords = [profaneWords];
+        }
+        asyncOutputWarning(
+          i18n.textToSpeechProfanity({
+            profanityCount: profaneWords.length,
+            profaneWords: profaneWords.join(', ')
+          })
+        );
+        Sounds.getSingleton().registerTextBytes(
+          language,
+          text,
+          true,
+          profaneWords
+        );
+        resolve('profanity found');
```

Comment:
How is this string used? Does it need to be translated?

Reply:
this is used to see if we resolved because we have actual sound or if we found profanity so this speech should be skipped over https://github.com/code-dot-org/code-dot-org/pull/35571/files#diff-c0ae5ae2bfb2b0d0c38c6345c9027e83R422

Reply:
Oh, I see. In that case can you make it a `const PROFANITY_FOUND` or similar? So that it's clear where it's coming from/how it's used/not just a string literal in multiple places?